src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs

```diff
@@ -377,38 +379,284 @@ internal PhysicalLocationVersionOne CreatePhysicalLocation(Region v2Region)
+                    }
+                    else if (v2Region.EndColumn > 0)
+                    {
+                        region.EndColumn = v2Region.EndColumn;
+                    }
+                    else
+                    {
+                        // THIS IS A PROBLEM. IF ALL THE "END" PROPERTIES ARE MISSING,
+                        // IT MEANS "THE REST OF THE StartLine". BUT IF CHARLENGTH IS
+                        // PRESENT BUT 0, IT MEANS "INSERTION POINT". AND WE CAN'T TELL
+                        // THE DIFFERENCE.
+                        // TODO: Issue #932
+
+                        // Assume it's an insertion point
+                        region.EndLine = region.EndColumn = region.Length = 0;
+                    }
+                }
+                else
+                {
+                    // There are no text-related properties. Therefore either the region is
+                    // described entirely by binary-related properties, or the region is an
+                    // insertion point at the start of the file.
+                    region.Length = v2Region.ByteLength;
+                    region.Offset = v2Region.ByteOffset;
+                }
             }
 
             return region;
         }
 
+        private int ConvertCharOffsetToByteOffset(int charOffset, Uri uri)
+        {
+            int byteOffset = 0;
+            Encoding encoding;
+
+            using (StreamReader reader = GetFileStreamReader(uri, out encoding))
+            {
+                if (reader != null)
+                {
+                    char[] buffer = new char[charOffset];
+
+                    // Read everything up to charOffset
+                    if (reader.ReadBlock(buffer, 0, buffer.Length) > 0)
+                    {
+                        byteOffset = encoding.GetByteCount(buffer, 0, buffer.Length);
+                    }
+                }
+            }
+
+            return byteOffset;
+        }
+
+        private int GetRegionByteLength(Region v2Region, Uri uri)
+        {
+            int byteLength = 0;
+            Encoding encoding;
+
+            using (StreamReader reader = GetFileStreamReader(uri, out encoding))
+            {
+                if (reader != null)
+                {
+                    if (v2Region.StartLine > 0) // Use line and column 
+                    {
+                        string sourceLine = null;
+
+                        // Read down to startLine (null return means EOF)
+                        for (int i = 1; i <= v2Region.StartLine && sourceLine != null; sourceLine = reader.ReadLine(), i++) { }
+
+                        if (sourceLine != null && sourceLine.Length <= v2Region.StartColumn)
+                        {
+                            // Since we read past startColumn, we need to back up using the base stream
+                            Stream stream = reader.BaseStream;
+                            stream.Position -= v2Region.StartColumn - 1;
```

Comment:
Bug: You need to back up by the full `StartColumn` characters. BUT....

Bug: `stream.Position` is measured in bytes, not characters.

Reply:
Ugh

Reply:
Bug: _If_ you end up using some variant of this method, be aware that `StartColumn` might be 0, in which case you'll need to set it to 1 (using a temporary variable, since I don't think you want to modify `v2Region`):
```
int startColumn = v2Region.StartColum > 0 ? v2Region.StartColumn : 1;
```

Reply:
I believe I have the logic wrong here. Should back up sourceLine.Length - (v2Region.StartColumn - 1) characters